src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/Repository/MethodologyRepositoryTests.cs

```diff
@@ -250,12 +251,247 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
             MockUtils.VerifyAllMocks(methodologyParentRepository);
         }
 
+        [Fact]
+        public async Task GetLatestPublishedByMethodologyParent()
+        {
+            var previousVersion = new Methodology
+            {
+                Id = Guid.Parse("7a2179a3-16a2-4eff-9be4-5a281d901213"),
+                PreviousVersionId = null,
+                PublishingStrategy = Immediately,
+                Status = Approved,
+                Version = 0
+            };
+
+            var latestPublishedVersion = new Methodology
+            {
+                Id = Guid.Parse("926750dc-b079-4acb-a6a2-71b550920e81"),
+                PreviousVersionId = Guid.Parse("7a2179a3-16a2-4eff-9be4-5a281d901213"),
+                PublishingStrategy = Immediately,
+                Status = Approved,
+                Version = 1
+            };
+
+            var latestDraftVersion = new Methodology
+            {
+                Id = Guid.Parse("9108ed11-ab53-4578-9e9a-f5cfc3443e66"),
+                PreviousVersionId = Guid.Parse("926750dc-b079-4acb-a6a2-71b550920e81"),
+                PublishingStrategy = Immediately,
+                Status = Draft,
+                Version = 2
+            };
+
+            var methodologyParent = new MethodologyParent
+            {
+                Versions = AsList(previousVersion, latestPublishedVersion, latestDraftVersion)
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.MethodologyParents.AddAsync(methodologyParent);
+                await contentDbContext.PublicationMethodologies.AddAsync(
+                    new PublicationMethodology
+                    {
+                        Publication = new Publication
+                        {
+                            Published = DateTime.UtcNow
+                        },
+                        MethodologyParent = methodologyParent,
+                        Owner = true
+                    });
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var methodologyParentRepository = new Mock<IMethodologyParentRepository>(MockBehavior.Strict);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                contentDbContext.AttachRange(methodologyParent);
```

Comment:
I've removed it now and in a few other cases.

I thought it was needed because of the call to lazily load the `Versions` collection of the parent.

The `Attach` calls are still needed in several unit tests in this class for testing `GetLatestByPublication` and `GetLatestPublishedByPublication`. This is because there is a the call to lazily load the `Versions` collection of the parent without the code every querying the parent from the context first. 

Where the parent is only ever returned by the mock, trying to lazily load the Versions property fails because the parent isn't being tracked yet.

We force it to be tracked by calling attach.

Reply:
Yep I've seen it in action before thanks!  Was nifty!